src/executor/hash_join.rs

```diff
@@ -38,6 +38,7 @@ impl HashJoinExecutor {
         // helper functions
         let left_rows = || left_chunks.iter().flat_map(|chunk| chunk.rows());
         let right_rows = || right_chunks.iter().flat_map(|chunk| chunk.rows());
+        let data_types = || self.left_types.iter().chain(self.right_types.iter());
```

Comment:
Now the closure can be removed?
```suggestion
        let data_types = self.left_types.iter().chain(self.right_types.iter());
```